src/pages/AddResult/AddResult.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from 'react';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { omitBy } from 'lodash';
+
+import { resultSchema } from 'utils/validationSchema';
+import { IApiResponse, ICreateResult, ISubject } from 'utils/types';
+import { grades } from 'utils/grades';
+import { createResult, updateResult } from 'api/result';
+import { fetchSubjects } from 'api/subject';
+import SelectComponent from 'components/SelectComponent';
+import showToast from 'utils/toastMessage';
+import Input from 'components/Input';
+import 'pages/AddResult/style.css';
+
+function AddResult() {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const existingData = location.state?.student;
+  const isUpdate = location.state?.isUpdate || false;
+  const [subjects, setSubjects] = useState<{ value: string; label: string }[]>([]);
+
+  const { register, handleSubmit, formState: { errors }, setValue } = useForm<ICreateResult>({
+    resolver: yupResolver(resultSchema),
+    mode: 'onBlur',
+    defaultValues: { ...existingData, subjectId: existingData?.subjectId._id } || {}
+  });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const subjectsResponse = await fetchSubjects();
+        if (subjectsResponse.data) {
+          const formattedSubjects = subjectsResponse.data.map((subject: ISubject) => ({
+            value: subject._id,
+            label: subject.subjectName,
+          }));
+          setSubjects(formattedSubjects);
+        }
+      } catch (error: IApiResponse | any) {
+        showToast(error);
+      }
+    };
+
+    fetchData();
```

Comment:
handle everything in a function so in future we just replace it by action

Reply:
Will be shift to reducer.